### ISSUE TITLE
Support for updating and checking golden files

### ISSUE THREAD

**Issue by swankjesse** [cashapp/paparazzi#34]
So far we’ve been thinking mostly about interactive development, but there are really 3 modes that Paparazzi can run in:

1. Development – emit screenshots for manual checking
2. Updating/creating golden files – generate screenshots intended for checking in
3. Checking golden files – confirm screenshots match golden files

I think the programmatic API should be the same in each mode: call snapshot().

But we should have mechanisms to create and check golden files to defend against regressions. Maybe a system property?
`-DpaparazziMode=dev`
`-DpaparazziMode=update`
`-DpaparazziMode=check`

The update mode should be opinionated about where golden files go. Perhaps something like this:

```
src/test/paparazzi/com/squareup/paparazzi/sample/KeypadViewTest.testViews.zero_dollars.png
src/test/paparazzi/com/squareup/paparazzi/sample/KeypadViewTest.testViews.five_bucks.png
src/test/paparazzi/com/squareup/paparazzi/sample/KeypadViewTest.testViews.grey.png
src/test/paparazzi/com/squareup/paparazzi/sample/KeypadViewTest.testViews.bolt.png
```

I’d love it if we didn’t actually need to write files to the filesystem in `check` mode. We already have a bitmap in memory.

When the check mode fails we should write a bitmap with actual image.

I’d also like it if we defer failure until after the test completes. That way we get all the failing screenshots.

**Comment by nikwotton** [comment]
bump. Also I'd like to request an api function that takes in a directory to put screenshots in, rather than hardcoding something. Maybe another system property like `-DgoldenFileDir=images/`?

**Comment by jrodbx** [comment]
Closed in #140
